lib/util/cache/package/sqlite.ts

```diff
@@ -0,0 +1,100 @@
+import Sqlite from 'better-sqlite3';
+import type { Database, Statement } from 'better-sqlite3';
+import * as upath from 'upath';
+import { logger } from '../../../logger';
+import { ensureDir } from '../../fs';
+
+export class SqlitePackageCache {
+  private readonly upsertStatement: Statement<unknown[]>;
+  private readonly getStatement: Statement<unknown[]>;
+  private readonly cleanupStatement: Statement<unknown[]>;
+  private readonly countStatement: Statement<unknown[]>;
+
+  static async init(cacheDir: string): Promise<SqlitePackageCache> {
+    const sqliteDir = upath.join(cacheDir, 'renovate/renovate-cache-sqlite');
+    await ensureDir(sqliteDir);
+    const sqliteFile = upath.join(sqliteDir, 'db.sqlite');
+    const client = new Sqlite(sqliteFile);
+    const res = new SqlitePackageCache(client);
```

Comment:
Does this create the file if it doesn't exist, and reuse the file if it does?

Reply:
Yes